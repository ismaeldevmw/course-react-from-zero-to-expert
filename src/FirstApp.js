import React from 'react';
import PropTypes from 'prop-types';
// import React, { Fragment } from 'react'

// Functional Components
const FirstApp = ({ greeting, subtitle }) => {  

    return (
    <>
        <h1>{ greeting }</h1>      
        <p> { subtitle } </p>
    </>
    );

}

FirstApp.propTypes = {
    greeting: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'Im a subtitle'
}

export default FirstApp;
