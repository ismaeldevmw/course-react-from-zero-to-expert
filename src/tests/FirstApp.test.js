import React from 'react'
// import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import FirstApp from '../FirstApp';


describe('Tests into <FirstApp />', () => {

  // test('should show the message "Hello! I am Goku"', () => {  
  //   const greeting = 'Hello! I am Goku';
  //   const { getByText } =  render( <FirstApp greeting={ greeting } /> );
  //   expect( getByText(greeting) ).toBeInTheDocument();
  // });

    test('should show <FirstApp /> properly', () => {

        const wrapper = shallow( <FirstApp greeting="Hello! I am Goku" /> );

        expect( wrapper ).toMatchSnapshot();

    });

});