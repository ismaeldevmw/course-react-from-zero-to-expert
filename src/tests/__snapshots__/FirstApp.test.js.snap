// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests into <FirstApp /> should show <FirstApp /> properly 1`] = `
<Fragment>
  <h1>
    Hello! I am Goku
  </h1>
  <p>
     
    Im a subtitle
     
  </p>
</Fragment>
`;
